### ISSUE TITLE
jpm sign: clean up the temporarily created XPI file (on error)

### ISSUE THREAD

**Issue by kumar303** [mozilla-jetpack/jpm#439]
If `jpm sign` encounters an error, it should remove the created XPI (if one was created). I think https://github.com/mozilla-jetpack/jpm/issues/398 just wasn't implemented for error cases.

**Comment by kumar303** [comment]
I opened this based on https://github.com/mozilla/olympia/issues/1106#issuecomment-165604233 but actually I think the XPI is cleaned up after the error. I think the original commentor mistook the temp XPI for a signed one.

**Comment by nikolas** [comment]
Yeah, it looks like it is a temporary XPI that would have been removed if my `jpm sign` command was successful.

**Comment by kumar303** [comment]
oh! Right. The file still exists but it's in the `/tmp`. We could add some cleanup to remove the tmp dir on errors to help avoid confusion.
